x-pack/plugins/security_solution/public/common/components/exceptions/viewer/index.tsx

```diff
@@ -4,96 +4,384 @@
+    (id: string | null): void => {
+      dispatch({
+        type: 'updateExceptionToDelete',
+        id,
+      });
+    },
+    [dispatch]
+  );
+
+  const onFetchList = useCallback(
+    ({ id, namespaceType }: ApiProps): void => {
+      if (fetchList != null && id != null && namespaceType != null) {
+        fetchList({
+          listId: id,
+          listNamespaceType: namespaceType,
+        });
+      }
+    },
+    [fetchList]
+  );
+
+  const onFiltersChange = useCallback(
+    (filter: { filter: string; tags: string[] }): void => {
+      dispatch({
+        type: 'updateFilterOptions',
+        filterOptions: filter,
+      });
+    },
+    [dispatch]
+  );
+
+  const onAddExceptionItem = useCallback((): void => {
+    setIsModalOpen(true);
+  }, [setIsModalOpen]);
 
+  const onEditExceptionItem = useCallback(
+    (exception: ExceptionListItemSchema): void => {
+      dispatch({
+        type: 'updateExceptionToEdit',
+        exception,
+      });
+
+      setIsModalOpen(true);
+    },
+    [setIsModalOpen]
+  );
+
+  const onCloseExceptionModal = useCallback(
+    ({ actionType, listId, listNamespaceType }): void => {
+      setIsModalOpen(false);
+
+      // TODO: This callback along with fetchList can probably get
+      // passed to the modal for it to call itself maybe
+      if (actionType === ModalAction.CREATE && listId != null && onAssociateList != null) {
+        onAssociateList(listId);
+      }
+
+      onFetchList({ id: listId, namespaceType: listNamespaceType });
+    },
+    [setIsModalOpen, onFetchList, onAssociateList]
+  );
+
+  const onDeleteException = useCallback(
+    async ({ id, namespaceType }: { id: string; namespaceType: NamespaceType }) => {
+      try {
+        const deleteTask = new AbortController();
+        setExceptionToDelete(id);
+        await deleteExceptionListItemById({ http, id, namespaceType, signal: deleteTask.signal });
+        setExceptionToDelete(null);
+
+        onFetchList({ id: selectedListId, namespaceType: selectedListNamespaceType });
+      } catch (error) {
+        onDispatchToaster({
+          color: 'danger',
+          title: i18n.DELETE_EXCEPTION_ERROR,
+          iconType: 'alert',
+        });
+        setExceptionToDelete(null);
+      }
+    },
+    [
+      dispatch,
+      http,
+      setExceptionToDelete,
+      onDispatchToaster,
+      selectedListNamespaceType,
+      selectedListId,
+    ]
+  );
+
+  const getListTypes = ({
+    detectionsRuleExcList,
+    endpointRuleExcList,
+  }: {
+    detectionsRuleExcList: RuleExceptionList | undefined;
+    endpointRuleExcList: RuleExceptionList | undefined;
+  }): ToggleId[] => {
+    if (detectionsRuleExcList == null) {
+      setSelectedListType(ToggleId.ENDPOINT);
+      return [ToggleId.ENDPOINT];
+    } else if (endpointRuleExcList == null) {
+      setSelectedListType(ToggleId.DETECTION_ENGINE);
+      return [ToggleId.DETECTION_ENGINE];
+    } else {
+      setSelectedListType(ToggleId.DETECTION_ENGINE);
```

Comment:
Is this going to make it so that we only show Detection Engine exception items by default? If so, I wonder it might be better to show items from both lists by default. Thoughts @spong ?

Reply:
It'd probably be good to switch from a `ButtonGroup` to a [`FilterGroup`](https://elastic.github.io/eui/#/forms/filter-group) here, and allow for a three-way toggle. On Detection Rules we can default to `Detection` being selected, and for promotion rules (or any rule with both types of lists) we can default to no selection which would then show all. This would behave just as the Elastic/Custom rules filter on the `All Rules` table. Adding the counts would be a nice addition as well 🙂 

<p align="center">
  <img width="300" src="https://user-images.githubusercontent.com/2946766/83813738-36468180-a67b-11ea-8f6c-b78fcbbf14ac.png" />
</p>

Reply:
It was on my list of TODOs to switch to `FilterGroup`! 😄  I'll update the logic to allow for this.

Reply:
Added! Now uses FilterGroup to allow user to toggle between all exception items, detections, or endpoint exceptions.